### PR TITLE
API for timelines without creating a new topic

### PR DESCRIPTION
**Pull request by rcillo**
This API change makes it possible to connect an event type to an
existing topic.

I considered that the timeline is the structure that acts as an interface to a low level storage and not the event type.

Adding a topic to the event type would not be enough because we would also need to specify the "storage". The only entity that is connected to a storage is the timeline so for me it was the logical place to do that. Also, attaching an event type to a topic eliminates the possibility to later on "attach" other topics to an event type.

So I think that allowing for this in the timeline is the most flexible place.

### CODE REVIEW

**Comment by adyach** [comment]
👍

File: src/acceptance-test/java/org/zalando/nakadi/webservice/TimelinesControllerAT.java [hunk 1]
```diff
@@ -1,59 +1,117 @@
...
+                .body("[1].event_type", equalTo(eventType.getName()))
                 .body("[1].order", Matchers.is(2))
-                .body("[1].storage_id", Matchers.equalTo("default"))
+                .body("[1].storage_id", equalTo("default"))
                 .body("[1].switched_at", dateWithin(1, MINUTES, new DateTime()))
                 .body("[1].cleaned_up_at", nullValue());
     }
 
+    @Test
+    public void testCreatTimelineOnExistentTopic() throws Exception {
```
Comment (lmontrieux) [inline src/acceptance-test/java/org/zalando/nakadi/webservice/TimelinesControllerAT.java]: Existing topic :)

File: src/acceptance-test/java/org/zalando/nakadi/webservice/TimelinesControllerAT.java [hunk 2]
```diff
@@ -1,59 +1,117 @@
...
+        final EventType eventType = EventTypeTestBuilder.builder()
+                .cleanupPolicy(CleanupPolicy.COMPACT)
+                .category(EventCategory.BUSINESS)
+                .enrichmentStrategies(ImmutableList.of(EnrichmentStrategyDescriptor.METADATA_ENRICHMENT))
+                .build();
+
+        // This event type is created only so that it creates a topic which we will use as an existent topic for the
```
Comment (lmontrieux) [inline src/acceptance-test/java/org/zalando/nakadi/webservice/TimelinesControllerAT.java]: existing

File: src/acceptance-test/java/org/zalando/nakadi/webservice/TimelinesControllerAT.java [hunk 3]
```diff
@@ -1,59 +1,117 @@
...
+        // new timeline. That's not how this feature will work on production, since topics are not shared between
```
Comment (lmontrieux) [inline src/acceptance-test/java/org/zalando/nakadi/webservice/TimelinesControllerAT.java]: in production

File: src/acceptance-test/java/org/zalando/nakadi/webservice/TimelinesControllerAT.java [hunk 4]
```diff
@@ -1,59 +1,117 @@
...
+        // KafkaRepository to create a dummy topic.
+        final EventType dummyEventType = EventTypeTestBuilder.builder()
+                .cleanupPolicy(CleanupPolicy.COMPACT)
+                .category(EventCategory.BUSINESS)
+                .enrichmentStrategies(ImmutableList.of(EnrichmentStrategyDescriptor.METADATA_ENRICHMENT))
+                .build();
+
+        NakadiTestUtils.createEventTypeInNakadi(eventType);
+        NakadiTestUtils.createEventTypeInNakadi(dummyEventType);
+
```
Comment (lmontrieux) [inline src/acceptance-test/java/org/zalando/nakadi/webservice/TimelinesControllerAT.java]: Why are there 2 blank lines here?

**Comment by rcillo** [comment]
We are no longer merging these changes since they are not necessary. We are not going to use KSQL anymore, so that we regain control over the topic creation.

**Pull request closed without merging**